ibm/service/mqcloud/resource_ibm_mqcloud_keystore_certificate.go

```diff
@@ -116,6 +122,38 @@ func ResourceIbmMqcloudKeystoreCertificate() *schema.Resource {
 				Computed:    true,
 				Description: "The URL for this key store certificate.",
 			},
+			"config": {
+				Type:        schema.TypeList,
+				Required:    true,
```

Comment:
Introducing Required field is a breaking change .
We are just reading "config" in read Method?

Reply:
`config` is an updating parameter inside this resource. so we usually do read and update operation on this property.

1. So if i am giving it as `Optional` field and trying to set `Default` value, i encounter the following error

```
│ Internal validation of the provider failed! This is always a bug
│ with the provider itself, and not a user issue. Please report
│ this bug:
│ 
│ 1 error occurred:
│       * resource ibm_mqcloud_keystore_certificate: config: Default is not valid for lists or sets
```

2. When i set config as `Optional field without Default `value an unwanted update operation appears on `terraform plan`
<img width="1016" alt="image" src="https://github.com/IBM-Cloud/terraform-provider-ibm/assets/42908782/7203fdb0-2a63-4fa4-b21e-6fc3eda84940">

3. As a result, I decided to set config as a `required` field.



Lemme know if i have any other option?

Reply:
Also, are there any problems or additional processes involved if there are breaking changes?

Reply:
4. Another option was to set this config property as both optional and computed. In that case, most of the operations worked fine. However, there is a state mismatch issue when we omit the config from the Terraform configuration; it does not default to an empty list.
<img width="1198" alt="image" src="https://github.com/IBM-Cloud/terraform-provider-ibm/assets/42908782/8b063b16-215f-42e7-8438-9a1938632447">

Reply:
Resolved by making the config property `optional` with some tweaking in the Read operation (see latest commit)